src/Geom_Box.f90

```diff
@@ -555,11 +574,22 @@ ELEMENTAL FUNCTION inside_ABBoxType(thisABB,p) RESULT(inside)
       TYPE(PointType),INTENT(IN) :: p
       REAL(SRK),PARAMETER :: fuzz=1e-6_SRK
       LOGICAL(SBK) :: inside
+
       inside=.FALSE.
-      IF(p%dim > 1) inside=((p%coord(1) > thisABB%xMin-fuzz) .AND. &
-        (p%coord(1) < thisABB%xMax+fuzz) .AND. &
-        (p%coord(2) > thisABB%yMin-fuzz) .AND. &
-        (p%coord(2) < thisABB%yMax+fuzz))
+      IF(p%dim == 2 .OR. (p%dim == 3 .AND. .NOT.thisABB%is3D)) THEN
+        inside=((p%coord(1) > thisABB%xMin-fuzz) .AND. &
+          (p%coord(1) < thisABB%xMax+fuzz) .AND. &
+          (p%coord(2) > thisABB%yMin-fuzz) .AND. &
+          (p%coord(2) < thisABB%yMax+fuzz))
+      ELSEIF(p%dim == 3 .AND. thisABB%is3D) THEN
+        inside=((p%coord(1) > thisABB%xMax-fuzz) .AND. &
+          (p%coord(1) < thisABB%xMax+fuzz) .AND. &
+          (p%coord(2) > thisABB%yMin-fuzz) .AND. &
+          (p%coord(2) < thisABB%yMax+fuzz) .AND. &
+          (p%coord(3) > thisABB%zMin-fuzz) .AND. &
+          (p%coord(3) < thisABB%zMax+fuzz))
+      ENDIF
+
```

Comment:
So I would do this a little differently:
```
IF(p%dim > 1) THEN
  inside=((p%coord(1) > thisABB%xMin-fuzz) .AND. &
         (p%coord(1) < thisABB%xMax+fuzz) .AND. &
         (p%coord(2) > thisABB%yMin-fuzz) .AND. &
         (p%coord(2) < thisABB%yMax+fuzz))
 IF(p%dim == 3 .AND. thisBB%is3D) inside=(inside .AND. &
          (p%coord(3) > thisABB%zMin-fuzz) .AND. &
           (p%coord(3) < thisABB%zMax+fuzz))
ENDIF
```

It just cuts down duplicating a few lines of code.  Just a suggestion though.  I'm fine with the changes as made.